### ISSUE TITLE
Failling rule triggers a fatal error when no message stack is set

### ISSUE THREAD

**Issue by gsouf** [particle-php/Validator#83]
This makes a fatal error : 

```php
        $length = new \Particle\Validator\Rule\Length(5);
       $length->validate("four");
```

**Comment by berry-langerak** [comment]
Hi @gsouf !

Thanks for taking the time to report this issue, and the PR to fix it. I fail to see why anyone would ever instantiate a rule, and call validate() on it, without using Particle\Validator. After all, the above logic should be triggered as:

```php
$v = new Validator();
$v->required('field')->length(5);
```

Do you have any reason to deviate from that?

**Comment by gsouf** [comment]
Hi Berry,

Indeed I did use the rule as a standalone without using the validator. That's how i got the fatal error.

**Comment by berry-langerak** [comment]
@gsouf I understand that you did use it stand-alone, but my question is rather why you would do so? After all, it's trivial to use Validator like it's supposed to be used, and just not use the error messages. I feel it's important to note that if you use a "Rule" instance directly, this might get you into troubles later, because you're not using the interface of Validator, but using it's implementation details directly -- something we don't take into account when refactoring.

Basically, the question I want to answer is if we actually need this particular piece of code to be in our code-base, especially if it's not the intended usage. 

As a work-around, it's also possible to do this:

```php
$length = new \Particle\Validator\Rule\Length(5);
$length->setMessageStack(new MessageStack());
$length->validate("four");
```

But like I said, I would advise against it. What do you think @RickvdStaaij?

**Comment by gsouf** [comment]
Yes, the reason I firstly used it as a standealone it's because I wanted to take advantage of the rules and not of the whole library.
Actualy I end up using a message stack, because I needed it : 
https://github.com/gsouf/UForm/blob/3d9c7047c2560d0018953444606fc88a74af99d3/lib/UForm/Validator/Particle/RuleBridge.php#L35
But in a first try I didnt need to use the message stack.

I think it's important to remove any possible error from a library. Maybe throw an exception if there is no message stack ? But I think it would be better to remove the possible fatal error

**Comment by berry-langerak** [comment]
Hi @gsouf. I'm going to close your issue. If you ever run into the need to use a rule separately from the Validator itself again, we can of course fix it ;)
